driver/src/orderbook/streamed/orderbook.rs

```diff
@@ -82,10 +82,15 @@ fn filter_account_state(
 }
 
 impl StableXOrderBookReading for Orderbook {
-    fn get_auction_data(&self, index: U256) -> Result<(AccountState, Vec<Order>)> {
+    fn get_auction_data(&self, batch_id_to_solve: U256) -> Result<(AccountState, Vec<Order>)> {
         // TODO: Handle future batch ids for when we want to do optimistic solving.
         let state = self.create_state()?;
-        let (account_state, orders) = state.orderbook_for_batch(Batch::Future(index.low_u32()))?;
+        // `orderbook_for_batch` takes the index of the auction that is currently collecting orders and returns
+        // the orderbook for the batch index that is currently being solved. `get_auction_data` passed in the
+        // index for the auction that orders should be valid for (the one currently being solved). Thus we need
+        // to increment it.
+        let (account_state, orders) =
+            state.orderbook_for_batch(Batch::Future(batch_id_to_solve.low_u32() + 1))?;
```

Comment:
Here we make it so that we query the state for the batch that is currently open (which returns data for the batch that is currently being solved)

Reply:
I have a question, should this be 
```suggestion
            state.orderbook_for_batch(Batch::Future(batch_id_to_solve.low_u32()))?;
```

And then the change to `driver/src/orderbook/streamed/state.rs` be
```rust
Ok((self.account_state(batch_id + 1), self.orders(batch_id)))
```

From what I understand, the order needs to be valid in the previous batch (i.e. the batch to solve, i.e. the current batch - 1), and the balances at the end of the batch to solve (i.e. beginning of the current batch).

Reply:
This would be possible too, however there is some assertion logic in the orderbook that errors if we request a batch from the past. 
And since the batch we are passing here is the one that is currently being solved, the event based orderbook might have already seen an event that is from the batch currently collecting orders (index + 1) making this batch here a batch from the past and thus throwing.